base/linalg/uniformscaling.jl

```diff
@@ -127,9 +127,22 @@ inv(J::UniformScaling) = UniformScaling(inv(J.λ))
 *(J1::UniformScaling, J2::UniformScaling) = UniformScaling(J1.λ*J2.λ)
 *(B::BitArray{2}, J::UniformScaling) = *(Array(B), J::UniformScaling)
 *(J::UniformScaling, B::BitArray{2}) = *(J::UniformScaling, Array(B))
-*(A::AbstractMatrix, J::UniformScaling) = J.λ == 1 ? A : J.λ*A
-*(J::UniformScaling, A::AbstractVecOrMat) = J.λ == 1 ? A : J.λ*A
-
+function *(A::AbstractMatrix, J::UniformScaling)
+    if J.λ == 1
+        T = promote_op(*, eltype(A), eltype(J))
+        return copy_oftype(A, T)
+    else
+        return J.λ*A
```

Comment:
should this be right multiplication in case the element types aren't commutative?